node/pod.go

```diff
@@ -156,40 +156,25 @@ func (pc *PodController) handleProviderError(ctx context.Context, span trace.Spa
 	span.SetStatus(origErr)
 }
 
-func (pc *PodController) deletePod(ctx context.Context, namespace, name string) error {
+func (pc *PodController) deletePod(ctx context.Context, pod *corev1.Pod) error {
 	ctx, span := trace.StartSpan(ctx, "deletePod")
 	defer span.End()
-
-	pod, err := pc.provider.GetPod(ctx, namespace, name)
-	if err != nil {
-		if errdefs.IsNotFound(err) {
-			// The provider is not aware of the pod, but we must still delete the Kubernetes API resource.
-			return pc.forceDeletePodResource(ctx, namespace, name)
-		}
-		return err
-	}
-
-	// NOTE: Some providers return a non-nil error in their GetPod implementation when the pod is not found while some other don't.
-	if pod == nil {
-		// The provider is not aware of the pod, but we must still delete the Kubernetes API resource.
-		return pc.forceDeletePodResource(ctx, namespace, name)
-	}
-
 	ctx = addPodAttributes(ctx, span, pod)
 
-	var delErr error
-	if delErr = pc.provider.DeletePod(ctx, pod.DeepCopy()); delErr != nil && !errdefs.IsNotFound(delErr) {
-		span.SetStatus(delErr)
-		return delErr
+	err := pc.provider.DeletePod(ctx, pod.DeepCopy())
+	if errdefs.IsNotFound(err) {
+		// TODO: Figure out what to do here. Previously, we used to delete the pod non-gracefully from Kubernetes
+		// The problem is that if a pod was deleted, and then it is no longer in the provider, we end up deleting it
+		// from kubernetes due to no reason whatsoever.
+		return err
 	}
 
-	log.G(ctx).Debug("Deleted pod from provider")
-
-	if err := pc.forceDeletePodResource(ctx, namespace, name); err != nil {
+	if err != nil {
 		span.SetStatus(err)
 		return err
 	}
-	log.G(ctx).Info("Deleted pod from Kubernetes")
+
+	log.G(ctx).Debug("Deleted pod from provider")
```

Comment:
Why are we changing the logging level?

Reply:
We aren't. Github seems confused. If you use the patience diff algorithm:

```
-func (pc *PodController) deletePod(ctx context.Context, namespace, name string) error {
+func (pc *PodController) deletePod(ctx context.Context, pod *corev1.Pod) error {
        ctx, span := trace.StartSpan(ctx, "deletePod")
        defer span.End()
-
-       pod, err := pc.provider.GetPod(ctx, namespace, name)
-       if err != nil {
-               if errdefs.IsNotFound(err) {
-                       // The provider is not aware of the pod, but we must still delete the Kubernetes API resource.
-                       return pc.forceDeletePodResource(ctx, namespace, name)
-               }
-               return err
-       }
-
-       // NOTE: Some providers return a non-nil error in their GetPod implementation when the pod is not found while some other don't.
-       if pod == nil {
-               // The provider is not aware of the pod, but we must still delete the Kubernetes API resource.
-               return pc.forceDeletePodResource(ctx, namespace, name)
-       }
-
        ctx = addPodAttributes(ctx, span, pod)
 
-       var delErr error
-       if delErr = pc.provider.DeletePod(ctx, pod.DeepCopy()); delErr != nil && !errdefs.IsNotFound(delErr) {
-               span.SetStatus(delErr)
-               return delErr
+       err := pc.provider.DeletePod(ctx, pod.DeepCopy())
+       if errdefs.IsNotFound(err) {
+               // TODO: Figure out what to do here. Previously, we used to delete the pod non-gracefully from Kubernetes
+               // The problem is that if a pod was deleted, and then it is no longer in the provider, we end up deleting it
+               // from kubernetes due to no reason whatsoever.
+               return err
+       }
+
+       if err != nil {
+               span.SetStatus(err)
+               return err
        }
 
        log.G(ctx).Debug("Deleted pod from provider")
 
-       if err := pc.forceDeletePodResource(ctx, namespace, name); err != nil {
-               span.SetStatus(err)
-               return err
-       }
-       log.G(ctx).Info("Deleted pod from Kubernetes")
-
        return nil
 }
```